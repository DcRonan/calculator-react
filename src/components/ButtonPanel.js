import Button from './Button';

const ButtonPanel = () => (
  <section>
    <div className="calculator">
      <div className="group-one">
        <Button name="AC" />
        <Button name="+/-" />
        <Button name="%" />
        <Button name="รท" />
      </div>
    </div>
  </section>
);

export default ButtonPanel;
