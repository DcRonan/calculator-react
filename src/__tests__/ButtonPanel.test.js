import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

test('should render correctly', () => {
  const handleClick = () => '';

  const tree = renderer
    .create(<ButtonPanel name="AC" handleClick={handleClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
