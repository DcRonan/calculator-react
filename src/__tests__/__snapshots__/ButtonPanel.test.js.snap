// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<section>
  <div
    className="calculator ButtonPanel"
  >
    <div
      className="group-one group"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="group-two group"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="group-three group"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="group-four group"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="group-five group"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
